### ISSUE TITLE
Priority scheduling is broken

### ISSUE THREAD

**Issue by Spiralo-Idioide** [tvhong/incremental-reading#18]
## Describe the bug
After activating priority scheduling, errors show up in multiple contexts.

## To Reproduce
First Activating priority mode.
Then multiple things leads to an error window.

#### Opening the organizer. The organizer doesn't open after closing the error window

```
Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue to the add-on author.
Debug info:
Anki 2.1.60 (76d88073) Python 3.10.9 Qt 6.4.2 PyQt 6.4.2
Platform: Linux-5.15.94-1-MANJARO-x86_64-with-glibc2.37
Flags: frz=False ao=True sv=3
Add-ons, last update check: 2023-02-23 22:27:55
Add-ons possibly involved: ⁨Incremental Reading v4115 unofficial clone⁩

Caught exception:
Traceback (most recent call last):
  File "/home/pierre/.var/app/net.ankiweb.Anki/data/Anki2/addons21/999215520/schedule.py", line 65, in showDialog
    if not self._getCardInfo(self._deckId):
  File "/home/pierre/.var/app/net.ankiweb.Anki/data/Anki2/addons21/999215520/schedule.py", line 298, in _getCardInfo
    prio = note[self._settings['prioField']]
  File "/usr/lib/python3.10/site-packages/anki/notes.py", line 159, in __getitem__
    return self.fields[self._field_index(key)]
TypeError: list indices must be integers or slices, not NoneType

```

#### Importing a webpage and setting a priority. No card is created after closing the error window
```
Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue to the add-on author.
Debug info:
Anki 2.1.60 (76d88073) Python 3.10.9 Qt 6.4.2 PyQt 6.4.2
Platform: Linux-5.15.94-1-MANJARO-x86_64-with-glibc2.37
Flags: frz=False ao=True sv=3
Add-ons, last update check: 2023-02-23 22:27:55
Add-ons possibly involved: ⁨Incremental Reading v4115 unofficial clone⁩

Caught exception:
Traceback (most recent call last):
  File "/home/pierre/.var/app/net.ankiweb.Anki/data/Anki2/addons21/999215520/importer.py", line 140, in importWebpage
    deck = self._createNote(webpage.title.string, body, source, priority)
  File "/home/pierre/.var/app/net.ankiweb.Anki/data/Anki2/addons21/999215520/importer.py", line 99, in _createNote
    setField(note, self._settings['prioField'], priority)
  File "/home/pierre/.var/app/net.ankiweb.Anki/data/Anki2/addons21/999215520/util.py", line 105, in setField
    note.fields[index] = value
TypeError: list indices must be integers or slices, not NoneType

```

#### Trying to open Browse for an IR3 card that was already added. The card cannot be edited after closing the error window

```
Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue to the add-on author.
Debug info:
Anki 2.1.60 (76d88073) Python 3.10.9 Qt 6.4.2 PyQt 6.4.2
Platform: Linux-5.15.94-1-MANJARO-x86_64-with-glibc2.37
Flags: frz=False ao=True sv=3
Add-ons, last update check: 2023-02-23 22:27:55

Caught exception:
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/aqt/progress.py", line 118, in handler
    func()
  File "/usr/lib/python3.10/site-packages/aqt/utils.py", line 1188, in <lambda>
    self.editor.call_after_note_saved(lambda: func(self, *args, **kwargs))
  File "/usr/lib/python3.10/site-packages/aqt/browser/browser.py", line 549, in on_all_or_selected_rows_changed
    self.editor.set_note(self.card.note(), focusTo=self.focusTo)
  File "/usr/lib/python3.10/site-packages/aqt/editor.py", line 487, in set_note
    self.loadNote(focusTo=focusTo)
  File "/usr/lib/python3.10/site-packages/aqt/editor.py", line 500, in loadNote
    for fld, val in self.note.items()
  File "/usr/lib/python3.10/site-packages/anki/notes.py", line 150, in items
    return [(f["name"], self.fields[ord]) for ord, f in sorted(self._fmap.values())]
TypeError: '<' not supported between instances of 'NoneType' and 'int'

```

#### Trying to extract a text when reviewing a card. No card is created after closing the error window

```
Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue to the add-on author.
Debug info:
Anki 2.1.60 (76d88073) Python 3.10.9 Qt 6.4.2 PyQt 6.4.2
Platform: Linux-5.15.94-1-MANJARO-x86_64-with-glibc2.37
Flags: frz=False ao=True sv=3
Add-ons, last update check: 2023-02-23 22:27:55
Add-ons possibly involved: ⁨Incremental Reading v4115 unofficial clone⁩

Caught exception:
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/aqt/webview.py", line 604, in handler
    cb(val)
  File "/home/pierre/.var/app/net.ankiweb.Anki/data/Anki2/addons21/999215520/text.py", line 78, in <lambda>
    'getHtmlText()', lambda text: self.create(text, settings)
  File "/home/pierre/.var/app/net.ankiweb.Anki/data/Anki2/addons21/999215520/text.py", line 132, in create
    getField(currentNote, settings['prioField']),
  File "/home/pierre/.var/app/net.ankiweb.Anki/data/Anki2/addons21/999215520/util.py", line 98, in getField
    return note.fields[index]
TypeError: list indices must be integers or slices, not NoneType

```

## Expected behavior
No error should show up and the card should be created.

## Specs
 - OS: Manjaro Linux. Kernel 5.15
 - Anki Version : 2.1.60 (but also present in 2.1.58). Both from the AUR
- Incremental Reading Version : v4.11.15

## Anki debug info

```
    
    Anki 2.1.60 (76d88073) Python 3.10.9 Qt 6.4.2 PyQt 6.4.2
    Platform: Linux-5.15.94-1-MANJARO-x86_64-with-glibc2.37
    Flags: frz=False ao=True sv=3
    Add-ons, last update check: 2023-02-23 22:27:55
    
    
    ===Add-ons (active)===
    (add-on provided name [Add-on folder, installed at, version, is config changed])
    AnkiConnect ['2055492159', 2022-09-19T02:11, 'None', mod]
    Incremental Reading v4115 unofficial clone ['999215520', 2023-02-20T21:57, 'None', '']
    
    ===IDs of active AnkiWeb add-ons===
    2055492159 999215520
```

**Comment by Spiralo-Idioide** [comment]
Posted the first message by error so I forgot to say that I am incredibly thankful that you are keeping this add-on alive. Manual scheduling is not the best solution, but at least it works and I can use IR.

So really thank you

**Comment by tvhong** [comment]
Hi,
Thank you for reporting the issue! I might have broken something while making other changes.

I'll try to get to this within the next week (and try add some tests to prevent regressions).

**Comment by tvhong** [comment]
Fixed in v4.11.6.
